### PR TITLE
RAD-1 Add sdf and archive_catalog sections

### PR DESCRIPTION
**Pull request by gardner358**
This is just my best guess.  For archive catalog I used the datamodelname but something I added the section.

sdf - I made lots of guesses.

### CODE REVIEW

File: src/rad/resources/schemas/program-1.0.0.yaml [hunk 1]
```diff
@@ -9,19 +9,54 @@ properties:
...
+    archive_catalog:
+      datatype: nvarchar(200) 
+      destination: [ScienceCommon.program_title]
   pi_name:
     title: Principle Investigator pi_name
     type: string
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: "= apt('./ProposalInformation/PrincipalInvestigator/InvestigatorAddress/LastName') + ', ' + apt('./ProposalInformation/PrincipalInvestigator/InvestigatorAddress/FirstName')"
```
Comment (perrygreenfield) [inline src/rad/resources/schemas/program-1.0.0.yaml:25]: A long line like this should be broken up. If you want it returned as a line without newlines you can use the following syntax (I'm lazy and haven't tested it myself :-).
`
origin: >
  "= apt('./ProposalInformation/PrincipalInvestigator/InvestigatorAddress/LastName')
   + ', ' + apt('./ProposalInformation/PrincipalInvestigator/InvestigatorAddress/FirstName')"
 

 
A long line like this should be broken up. If you want it returned as a line without newlines you can use the following syntax (I'm lazy and haven't tested it myself :-).
```
origin: >
  "= apt('./ProposalInformation/PrincipalInvestigator/InvestigatorAddress/LastName')
  + ', ' + apt('./ProposalInformation/PrincipalInvestigator/InvestigatorAddress/FirstName')"
```

**Review by perrygreenfield — COMMENTED** [review]
These generally look good to me, only one change due to a long line is needed. 

Though there is another issue that must be dealt with in that my latest PR makes changes that you will have to fold in, but it shouldn't take a lot of work, nothing like what you had to do for this.

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 2]
```diff
@@ -12,84 +12,273 @@ properties:
   gs_start_time:
     title: UTC time when guide star activity started
     type: string
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: TBD
+    archive_catalog:
+      datatype: datetime2
```
Comment (PaulHuwe) [inline src/rad/resources/schemas/guidestar-1.0.0.yaml:20]: What is the difference between datetime and datetime2?

Comment (gardner358) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:20]: datetime has up to 3 milliseconds and datetime2 has up to 7 milliseconds.  But datetime is eventually supposed to be deprecated so we use datetime2 in the archive catalog.

Comment (PaulHuwe) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:20]: As in 3 or 7 millisecond digits of precision?

Comment (gardner358) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:20]: Yes.

File: src/rad/resources/schemas/ephemeris-1.0.0.yaml [hunk 3]
```diff
@@ -9,41 +9,118 @@ properties:
   earth_angle:
     title: "Earth Angle [radians]"
     type: number
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: PSS:spacecraft_parameters.earth_avoidance_angle????
```
Comment (ddavis-stsci) [inline src/rad/resources/schemas/ephemeris-1.0.0.yaml:15]: I don't think these are avoidance angles but the angle between the telescope pointing and the sun, moon, or earth. They are calculated for the observation and are in PSS (or pps for JWST).

File: src/rad/resources/schemas/ephemeris-1.0.0.yaml [hunk 4]
```diff
@@ -9,41 +9,118 @@ properties:
...
     type: number
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: PSS:ephemeris_status.x
+    archive_catalog:
+      datatype: float
+      destination: [ScienceCommon.spatial_x]
   spatial_y:
     title: "[km] Y spatial coordinate of JWST"
```
Comment (ddavis-stsci) [inline src/rad/resources/schemas/ephemeris-1.0.0.yaml:70]: JWST --> Roman

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 5]
```diff
@@ -12,84 +12,273 @@ properties:
   gs_start_time:
     title: UTC time when guide star activity started
```
Comment (ddavis-stsci) [inline src/rad/resources/schemas/guidestar-1.0.0.yaml:13]: guide star --> guide window
gs --> gw?

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 6]
```diff
@@ -12,84 +12,273 @@ properties:
...
     type: number
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: TBD
+    archive_catalog:
+      datatype: float
+      destination: [ScienceCommon.gs_udec]
   gs_mag:
     title: guide star magnitude in FGS detector
```
Comment (ddavis-stsci) [inline src/rad/resources/schemas/guidestar-1.0.0.yaml:83]: FGS --> WFI or Wide field imager?

**Review by ddavis-stsci — APPROVED** [review]
A few minor points but overall it looks good.

File: src/rad/resources/schemas/ephemeris-1.0.0.yaml (hunk 3 above)
Comment (gardner358) [reply src/rad/resources/schemas/ephemeris-1.0.0.yaml:15]: I put these back to TBD.  More work will have to be done to find out where they are in PSS.

File: src/rad/resources/schemas/ephemeris-1.0.0.yaml (hunk 4 above)
Comment (gardner358) [reply src/rad/resources/schemas/ephemeris-1.0.0.yaml:70]: Done

File: src/rad/resources/schemas/guidestar-1.0.0.yaml (hunk 5 above)
Comment (gardner358) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:13]: I'm going to leave this as is now.  We are having a meeting to go over the the RAD schema files.  We'll talk about it at that meeting.

File: src/rad/resources/schemas/guidestar-1.0.0.yaml (hunk 6 above)
Comment (gardner358) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:83]: I'm going to leave this as is now.  We are having a meeting to go over the the RAD schema files.  We'll talk about it at that meeting.

**Pull request merged**